libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs

```diff
@@ -148,7 +150,8 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
 
             foreach (var caseScope in this.Cases)
             {
-                var (value, error) = this.caseExpressions[caseScope.Value].TryEvaluate(dc.State);
+                var opt = new Options() { Locale = dc.GetLocale() };
```

Comment:
If we change the options to default to Thread.CurrentCulture then there is no need to change all of these calls.

Reply:
I changed the default locale to Thread.CurrentThread.CurrentCulture.Name.
It will always use the default locale other the locale stored in the turn.locale.
I have a question for these actions and selectors.
Should the expressions used in these cases are locale independent?